DevEdu.API/DevEdu.Business.Tests/UserServiceTests.cs

```diff
@@ -0,0 +1,257 @@
+﻿using DevEdu.Business.Servicies;
+using DevEdu.DAL.Enums;
+using DevEdu.DAL.Models;
+using DevEdu.DAL.Repositories;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace DevEdu.Business.Tests.Services
+{
+    public class UserServiceTests
+    {
+
+        private Mock<IUserRepository> _userRepoMock;
+
+        [SetUp]
+        public void Setup()
+        {
+            _userRepoMock = new Mock<IUserRepository>();
+        }
+
+        [Test]
+        public void AddUser_DtoWithRoles_UserDtoWithRolesCreated()
+        {
+            //Given
+            var expectedUserId = 1;
+            var _userDto = new UserDto
```

Comment:
please create a static class (UserData) and move this mock to a static method

please do it for all the methods of this class